sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/EncoderUtils.scala

```diff
@@ -77,6 +77,7 @@ object EncoderUtils {
       case _: DecimalType => classOf[Decimal]
       case _: DayTimeIntervalType => classOf[PhysicalLongType.InternalType]
       case _: YearMonthIntervalType => classOf[PhysicalIntegerType.InternalType]
+      case _: StringType => classOf[UTF8String]
```

Comment:
why do we need this change now?

Reply:
This is required for `LuhnCheck`. It has a runtime replacement: https://github.com/apache/spark/blob/1676a4a5da0e431cce7b884001863a4bbe13b7d6/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L3290-L3295 The path from `StaticInvoke` eventually leads to this function which was falling back to Object type and the test was failing.